libraries/lib-wave-track/WaveClip.h

```diff
@@ -95,6 +95,13 @@ struct WaveClipListener : WaveClipListenerBase {
    virtual ~WaveClipListener() = 0;
    virtual void MarkChanged() = 0;
    virtual void Invalidate() = 0;
+
+   // Default implementation does nothing
+   virtual void WriteXMLAttributes(XMLWriter &writer) const;
```

Comment:
I think this violates SRP and in a bad way. I missed it somehow, during the first pass of the review, but it caught my eye when I saw color index serialization in the wave cache.

Reply:
I have a strong opinion that the fix is worse than the problem. I understand that it is pointless to argue.